File: comu/com.energyict.mdc.device.data.rest/src/main/java/com/energyict/mdc/device/command/rest/impl/DualControlInfo.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.device.command.rest.impl;

import com.elster.jupiter.rest.util.DualControlChangeInfo;

import java.util.List;

public class DualControlInfo {
    public boolean hasCurrentUserAccepted;
    public List<DualControlChangeInfo> changes;
    public PendingChangesType pendingChangesType;

    public DualControlInfo() {
    }
}

File: comu/com.energyict.comserver.comserver.core.extjs/src/main/web/js/mdc/src/model/DualControlInfo.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Mdc.model.DualControlInfo', {
    extend: 'Ext.data.Model',

    requires: [
        'Uni.model.PendingChange'
    ],

    fields: [
        {name: 'hasCurrentUserAccepted', type: 'boolean', defaultValue: true, persist: false},
        {name: 'pendingChangesType', type: 'auto'}
    ],

    associations: [
        {
            name: 'changes',
            type: 'hasMany',
            model: 'Uni.model.PendingChange',
            associationKey: 'changes',
            foreignKey: 'changes',
            getTypeDiscriminator: function (node) {
                return 'Uni.model.PendingChange';
            }
        }
    ]

});